book/src/concepts/entity_and_component.md

```diff
@@ -0,0 +1,105 @@
+```
+where u32 is the id of the entity and generation is used to check if the entity has been deleted.
+
+`Entity`s are stored in a special container `EntitiesRes`. Whereas the data associated with the entities are grouped into components and stored in the designated storages.
+
+Consider an example where you have three objects: two bottles and a person.
+
+* Bottle A: x: 150.0, y: 202.1, shape: "round", color: "red"
+
+* Bottle B: x: 15.0, y: 22.0, shape: "square", color: "white"
+
+* Person C: x: 100.5, y: 300.8, name: "Peter"
+
+We can separate bottle's properties into `PositionComponent` and `BottleComponent`, and person's properties into `PositionComponent` and `PersonComponent`. Here's an illustration of how the three objects would be stored.
+
+![How entity and components are stored](../images/concepts/component_and_entity.svg)
+
+As you could see from the graph, entities do not store data. Nor do they know any information about their components. They serve the purpose of object identification and tracking object existance. The components store all the data and their connection to entities. 
+
+If you know database systems, this organization looks quite similar to the tables in a database, where entity id serves as the key in each table. In fact, you could even join components or entities like joining tables. For example, to update the position of all the persons, you will need to join the PersonComponent and the PositionComponent. 
+
+## EntitiesRes
+
+Even though the structure of the entity is pretty simple, entity manipulation is very sophisticated and crucial to game preformance. This is why entities are handled exclusively by the struct `EntitiesRes`. `EntitiesRes` provides two ways for creating/deleting entities:
+
+* Immediate creation/deletion, used for game setup or clean up.
+* Atomic or delayed creation/deletion, used in the game play state. It updates entities in batch at the end of each game loop.
+
+You will see how these methods are used in later chapters.
+
+
+## Declaring a component
+
+To declare a component, you first declare the relevant underlying data:
+
+```rust,no_run,noplaypen
+/// This `Component` describes the shape of an `Entity`
+enum Shape {
+    Sphere { radius: f32 },
+    RectangularPrism { height: f32, width: f32, depth: f32 },
+}
+
+/// This `Component` describes the contents of an `Entity`
+pub struct Content {
+    content_name: String,
+}
+```
+
+and then you implement the `Component` trait for them:
+
+```rust,no_run,noplaypen
+# extern crate amethyst;
+# struct Shape;
+# struct Content;
+use amethyst::ecs::{Component, DenseVecStorage};
+
+impl Component for Shape {
+    type Storage = DenseVecStorage<Self>;
+}
+
+impl Component for Content {
+    type Storage = DenseVecStorage<Self>;
+}
+```
+Defining the Storage type will determine how you store the component, but it will not initiate the storage. Storage is initiated when you register a component in `World`. You will see it in the World chapter. 
```

Comment:
"The storage...". The sentence sounds strange without it - more like a enumeration or a note instead of a book.

"You will see how to do so...". "it" alone sounds strange imho